radix-engine/src/track/track.rs

```diff
@@ -29,6 +29,17 @@ pub struct StateUpdates {
 }
 
 impl StateUpdates {
+    /// Merges the updates from the given instance into this one.
+    pub fn extend(&mut self, mut other: Self) {
+        self.partition_deletions
+            .extend(other.partition_deletions.drain(..));
```

Comment:
Minor: I don't believe `drain()` is necessary

Reply:
ahhhh right; luckily, it vanishes in the successor PR anyways (https://github.com/radixdlt/radixdlt-scrypto/pull/1454/files#diff-3c2e40e4ace40858d7695c9dad2f13e421e22cb56b4691b2c62f9ef6d133cc0eL33)